fido2/client.py

```diff
@@ -578,3 +588,148 @@ def _ctap1_get_assertion(
                 if e.code == ClientError.ERR.TIMEOUT:
                     raise  # Other errors are ignored so we move to the next.
         raise ClientError.ERR.DEVICE_INELIGIBLE()
+
+
+_WIN_INFO = Info.create(["U2F_V2", "FIDO_2_0"])
+
+
+class WindowsClient(object):
+    """Fido2Client-like class using the Windows WebAuthn API.
+
+    Note: This class only works on Windows 10 19H1 or later. This is also when Windows
+    started restricting access to FIDO devices, causing the standard client classes to
+    require admin priveleges to run (unlike this one).
+
+    The make_credential and get_assertion methods are intended to work as a drop-in
+    replacement for the Fido2Client methods of the same name.
+
+    :param str origin: The origin to use.
+    :param verify: Function to verify an RP ID for a given origin.
+    :param ctypes.wintypes.HWND handle: (optional) Window reference to use.
+    """
+
+    def __init__(self, origin, verify=verify_rp_id, handle=None):
+        self.api = WinAPI(handle)
+        self.origin = origin
+        self._verify = verify
+
+    @property
+    def info(self):
+        return _WIN_INFO
+
+    @staticmethod
+    def is_available():
+        return platform.system().lower() == "windows" and WinAPI.version > 0
+
+    def _verify_rp_id(self, rp_id):
+        try:
+            if self._verify(rp_id, self.origin):
+                return
+        except Exception:
+            pass  # Fall through to ClientError
+        raise ClientError.ERR.BAD_REQUEST()
+
+    def make_credential(
+        self,
+        rp,
+        user,
+        challenge,
+        algos=[ES256.ALGORITHM],
+        exclude_list=None,
+        extensions=None,
+        rk=False,
+        uv=False,
+        pin=None,
+        timeout=None,
+        on_keepalive=None,
+    ):
+        """Create credentials using Windows WebAuhtN APIs.
+
+        Args:
+            client_data: ClientData
+            rp: Relying Party information
+            user: Information about the user
+            algos: Algorithms used
+            exclude_list: List of credentials to exclude
+            extensions: Any credential extensions
+            rk: Whether the Resident Key is required
+            uv: Whether or not User Verification is required
+            timeout: Timeout while creating credential
+        """
+
+        self._verify_rp_id(rp["id"])
+
+        client_data = ClientData.build(
+            type=WEBAUTHN_TYPE.MAKE_CREDENTIAL,
+            clientExtensions={},
+            challenge=challenge,
+            origin=self.origin,
+        )
+
+        result = self.api.make_credential(
+            rp,
+            user,
+            [{"type": "public-key", "alg": alg} for alg in algos],
+            client_data,
+            timeout or 0,
+            rk,
+            WebAuthNAuthenticatorAttachment.ANY,
```

Comment:
When the attachment of `ANY` is specified - it prompts for Windows login credentials and doesn't use the Yubikey.
![image](https://user-images.githubusercontent.com/627555/68600376-696fbc00-0470-11ea-95d5-7816d2506167.png)

Here it's asking for my pin to login to Windows (not the PIN of the Yubikey)

Reply:
Good catch I'll change it to cross-platform for now. I would like to make it specifiable by the caller in the end, still thinking about the best way to go about that.

Reply:
For make_credential this is now up to the caller as per the WebAuthn spec. For get_assertion this is still hard coded to ANY, but I don't believe this causes any issues in practice (please let me know if it does).